coding/javascript.md

```diff
@@ -37,6 +37,26 @@ We use [Yarn](https://yarnpkg.com/en) for dependency management.
 
 We use the [Jest](https://jestjs.io) unittest framework.
 
+#### General Testing Principals
+  * Avoid setting up state in `beforeEach`. It should be clear from reading a test in isolation what state it is in.
+  * Use test factories to reduce boilerplate for commonly instantiated objects.
```

Comment:
These first two points seem quite vague without the examples. What "state" we don't want to set in `beforeEach` (and what is OK to be set in there), what are "test factories", what should they produce?

Reply:
Sure, I'll try to come up with some examples.

Reply:
I think the "state" point could be a principle, as long as we define what "state" means.

The "factories" point though - this seems an inherently nuanced point about the choices of the individual developer, not significantly distinct from choices of encapsulation in general. Maybe this point should just be removed, until we have enough examples of people "doing it wrong" to distil it into a practice? Or maybe it should be moved to a wiki post or something about tricks that can help with testing. I'm just not sure the use of factories counts as a "principle".